### PR TITLE
add TargetOffset

### PR DESCRIPTION
**Pull request by Ruemena**
this adds targetoffset, a way to change the number of targets displayed to scps. this allows you to implement npcs, custom roles, etc that aren't displayed as targets (among other things)
TargetOffset is added to the total number of targets - it doesn't allow you to directly set the number of targets directly, as that would break compatibility between plugins
UpdateTargetCount() forces an update in the number of targets. this is necessary because you may not want to immediately update the number of targets displayed.

tested and works fine (for real)

**THIS MAY STILL NEED SOME REFINING** as rn there's no way to not have an incredibly quick increase followed by a decrease or vice versa when modifying the targetoffset and spawning or killing a player. this could prob be fixed by sending a sync var on the spawning event, or something like that

### CODE REVIEW

**Review by louis1706 — CHANGES_REQUESTED** [review]
the Patches should be ine Generic file not Fixed

File: Exiled.Events/Patches/Generic/TargetOffset.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// -----------------------------------------------------------------------
+// <copyright file="TargetOffset.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Player
```
Comment (louis1706) [inline Exiled.Events/Patches/Generic/TargetOffset.cs:8]: This is not the correct 'amespace

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Generic/TargetOffset.cs (hunk 1 above)
Comment (Ruemena) [reply Exiled.Events/Patches/Generic/TargetOffset.cs]: fixed

**Review by louis1706 — CHANGES_REQUESTED** [review]
Fix Build Error

File: Exiled.API/Features/Round.cs [hunk 2]
```diff
@@ -174,6 +174,23 @@ public static IEnumerable<Side> AliveSides
             }
         }
 
+        /// <summary>
+        /// Gets or sets a visual offset applied to the target counter for SCPs.
+        /// </summary>
+        /// <remarks>This does not update the target counter automatically. To do that, call Round.UpdateTargetCounter().</remarks>
+        public static int TargetOffset { get; set; } = 0;
```
Comment (louis1706) [inline Exiled.API/Features/Round.cs:181]:
```suggestion
        public static int TargetOffset { get; set; }
```

**Review by louis1706 — APPROVED** [review]

File: Exiled.API/Features/Round.cs [hunk 3]
```diff
@@ -239,5 +254,16 @@ public static bool EndRound(bool forceEnd = false)
         /// Start the round.
         /// </summary>
         public static void Start() => CharacterClassManager.ForceRoundStart();
+
+        /// <summary>
+        /// Updates the target counter for SCPs.
+        /// </summary>
+        private static void UpdateTargetCounter()
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Round.cs:261]: This method is called once, why making it a method then?

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Round.cs [hunk 4]
```diff
@@ -248,5 +263,16 @@ public static bool EndRound(bool forceEnd = false)
...
+        {
+            foreach (Player player in Player.List)
+            {
+                player.SendFakeSyncVar(RoundSummary.singleton.netIdentity, typeof(RoundSummary), nameof(RoundSummary.Network_chaosTargetCount), RoundSummary.singleton._chaosTargetCount + TargetOffset);
+            }
+        }
```
Comment (louis1706) [inline Exiled.API/Features/Round.cs:276]:
```suggestion
```

File: Exiled.API/Features/Round.cs [hunk 5]
```diff
@@ -183,6 +185,19 @@ public static IEnumerable<Side> AliveSides
...
+        /// <summary>
+        /// Gets or sets a visual offset applied to the target counter for SCPs.
+        /// </summary>
+        public static int TargetOffset
+        {
+            get => targetOffset;
+            set
+            {
+                targetOffset = value;
+                UpdateTargetCounter();
```
Comment (louis1706) [inline Exiled.API/Features/Round.cs:197]:
```suggestion
            foreach (Player player in Player.List)
            {
                player.SendFakeSyncVar(RoundSummary.singleton.netIdentity, typeof(RoundSummary), nameof(RoundSummary.Network_chaosTargetCount), RoundSummary.singleton._chaosTargetCount + TargetOffset);
            }
```

File: Exiled.API/Features/Round.cs [hunk 6]
```diff
@@ -183,6 +185,22 @@ public static IEnumerable<Side> AliveSides
...
+        public static int TargetOffset
+        {
+            get => targetOffset;
+            set
+            {
+                targetOffset = value;
+            foreach (Player player in Player.List)
+            {
+                player.SendFakeSyncVar(RoundSummary.singleton.netIdentity, typeof(RoundSummary), nameof(RoundSummary.Network_chaosTargetCount), RoundSummary.singleton._chaosTargetCount + TargetOffset);
+            }
```
Comment (louis1706) [inline Exiled.API/Features/Round.cs:200]:
```suggestion
                targetOffset = value;
                foreach (Player player in Player.List)
                {
                    player.SendFakeSyncVar(RoundSummary.singleton.netIdentity, typeof(RoundSummary), nameof(RoundSummary.Network_chaosTargetCount), RoundSummary.singleton._chaosTargetCount + TargetOffset);
                }
```

**Pull request merged**